app/models/like.rb

```diff
@@ -1,4 +1,20 @@
 class Like < ApplicationRecord
-  belongs_to :users
-  belongs_to :posts
+  belongs_to :user, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  def initialize(*args)
+    super(*args)
+    update_likes_counter if args.any? # Executes the method only if an argument is provided.
+  end
+
+  # Method to update the likes counter for a post.
+  def update_likes_counter
+    post = Post.find_by(id: post_id)
+    counter = if post.likes_counter.nil?
+                1
+              else
+                post.likes_counter + 1
+              end
+    post.update(likes_counter: counter)
+  end
```

Comment:
- [ ] I suggest that you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_likes_counter` method every time a new like has been saved to the database instead of using the initialize method. :relaxed: 
  
  ```ruby
  after_save :update_likes_counter
  ```
- [ ] I suggest that you use the `increment` method for the `likes_counter` attribute of a post object. This is useful if you want to keep track of the number of likes created by a post without having to recompute the count every time. 
  
  ```ruby
  def update_likes_counter
    post.increment!(:posts_counter)
  end
  ```